packages/core/src/__tests__/elementZipper.js

```diff
@@ -0,0 +1,113 @@
+'use strict';
+
+import { fromJS } from 'immutable';
+import { zip } from '..';
+
+describe('Zipper', () => {
+
+  const singleChild = {
+    kind: 'parent',
+    children: [
+      {
+        kind: 'lvl1',
+        children: [
+          {
+            kind: 'lvl2'
+          }
+        ]
+      }
+    ]
+  }
+
+  it('zipper should correctly navigate up and down', () => {
+    const zipper = zip.elementZipper(fromJS(singleChild), 'children')
+    expect(zipper.value().get('kind')).toEqual('parent')
+    expect(zipper.down().value().size).toEqual(1) // the children collection level
+    expect(zipper.down().down().value().get('kind')).toEqual('lvl1')
+    expect(zipper.down().down().down().value().size).toEqual(1) // the children collection level
+    expect(zipper.down().down().down().down().value().get('kind')).toEqual('lvl2')
+    expect(zipper.down().down().down().down().down()).toBeNull()
+    expect(zipper.down().up().value().get('kind')).toEqual('parent')
+    expect(zipper.down().down().down().up().value().get('kind')).toEqual('lvl1')
+  });
+
+  const multipleChildren = {
```

Comment:
One more thing here: test are documentation; here you have an element tree but elements have no *kind* property. This will confuse people.